pkg/auth/cache/synchronized_access_cache_test.go

```diff
@@ -57,6 +58,65 @@ var _ = Describe("SynchronizedAccessCache", func() {
 		subjectLocator = mocks.NewMockSubjectLocator(ctrl)
 	})
 
+	It("will timeout on long running synch", func(ctx context.Context) {
+		// given
+		namespaceLister := mocks.NewMockClientReader(ctrl)
+		namespaceLister.EXPECT().
+			List(gomock.Any(), gomock.Any(), gomock.Any()).
+			DoAndReturn(func(ctx context.Context, nn *corev1.NamespaceList, opts ...client.ListOption) error {
+				<-ctx.Done()
+				return ctx.Err()
+			}).
+			Times(1)
+		nsc := cache.NewSynchronizedAccessCache(subjectLocator, namespaceLister, cache.CacheSynchronizerOptions{
+			SynchTimeout: 1 * time.Second,
+		})
+
+		// when
+		Expect(nsc.Synch(ctx)).
+			// then
+			To(MatchError("context deadline exceeded"))
+	})
+
+	It("will return empty data if timed-out", Label("aaa"), func(ctx context.Context) {
+		// given
+		namespaceLister := mocks.NewMockClientReader(ctrl)
+		namespaceLister.EXPECT().
+			List(gomock.Any(), gomock.Any(), gomock.Any()).
+			DoAndReturn(func(ctx context.Context, nn *corev1.NamespaceList, opts ...client.ListOption) error {
+				(&corev1.NamespaceList{
+					Items: []corev1.Namespace{
+						{ObjectMeta: metav1.ObjectMeta{Name: "myns-1"}},
+						{ObjectMeta: metav1.ObjectMeta{Name: "myns-2"}},
+						{ObjectMeta: metav1.ObjectMeta{Name: "myns-3"}},
+					},
+				}).DeepCopyInto(nn)
+				return nil
+			}).
+			Times(1)
+
+		subjectLocator.EXPECT().
+			AllowedSubjects(gomock.Any()).
+			DoAndReturn(func(attributes authorizer.Attributes) ([]rbacv1.Subject, error) {
+				// reply fast the first time
+				return []rbacv1.Subject{userSubject}, nil
+			}).
```

Comment:
Okay, now that I read my own suggestion, there's an even better way to do this one:
```suggestion
			// reply fast the first time
			Return([]rbacv1.Subject{userSubject}, nil).
```

I think my brain must be malfunctioning today :laughing: